### PR TITLE
#46: typescript support (closes #46)

### LINKED ISSUE #46: typescript support
## requirements
add typescript support

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by veej**
Closes #46

### tests performed
VSCode can see the types

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+module.exports = require('lib');
```
Comment (FrancescoCioria) [inline index.js]: wait, this file is not being used (it's not in the `main` nor in `files`)

Comment (veej) [inline index.js]: whoops, I forget to update `main`

Comment (FrancescoCioria) [inline index.js]: you should also add `index.js` to the files otherwise 💣 💥

File: package.json [hunk 2]
```diff
@@ -73,6 +75,7 @@
     "react-dom": "^0.14.0 || ^15.0.0"
   },
   "dependencies": {
+    "@types/react": "^15.0.24",
```
Comment (FrancescoCioria) [inline package.json]: this should be a dev-dependency

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+
+export interface TextareaAutosizeProps extends React.HTMLAttributes<HTMLTextAreaElement> {
+  maxRows?: number,
+  onResize?: (e: React.SyntheticEvent<Event>) => void,
```
Comment (FrancescoCioria) [inline index.d.ts]: isn't this missing `rows`?

also I have a question, what happens if you pass another prop that is accepted by native `textarea` like `onClick` or `onFocus`, does it complain? (it shouldn't)

Comment (FrancescoCioria) [inline index.d.ts]: also, `onResize` does not return a native event but a custom one from the library `autosize`. Are you sure this is correct?

Comment (veej) [inline index.d.ts]: we're extending `React.HTMLAttributes<HTMLTextAreaElement>` to check those props. `rows` is a prop of `textarea`

Comment (veej) [inline index.d.ts]: We checked the library with @gcanti and we found that `autosize` is calling a `new Event(...)`, so it should be correct

**Pull request merged**